Testing/PerformanceTests/analysis.py

```diff
@@ -115,147 +129,85 @@ def get_results_matching(results, field, value):
-def plot_success_count(type='system', last_num=-1, x_field='revision'):
-    """ Plot the count of successful/failed tests vs revision number
-
-    Parameters
-    ----------
-        type :: 'system', or 'performance'
-    """
-    results = get_results('', type, where_clause='', orderby_clause=get_orderby_clause(last_num))
-    revisions = get_unique_fields(results, x_field)
-
-    # Go through each revision
-    success = []
-    fail = []
-    for revision in revisions:
-        these = get_results_matching(results, x_field, revision)
-        succeeded = 0
-        failed = 0
-        for res in these:
-            if res["success"]:
-                succeeded += 1
-            else:
-                failed += 1
-        # Keep the list of them
-        success.append(succeeded)
-        fail.append(failed)
-
-    figure()
-    revisions = np.array(revisions)
-    fail = np.array(fail)
-    success = np.array(success)
-
-    index = np.arange(len(revisions))
-#    p1 = bar(index, fail, color='r')
-#    p2 = bar(index, success, color='g', bottom=fail)
-#    legend( (p1[0], p2[0]), ('Failure', 'Success') )
-
-    p1 = fill_between(index, fail, 0, color='r')
-    p2 = fill_between(index, success+fail, fail, color='g')
-    #legend( (p1, p2), ('Failure', 'Success') )
-
-    smart_ticks( index, revisions)
-
-
-    ylabel('Success/Fail')
-    xlabel(x_field)
-    revsare = "all revs"
-    if last_num > 0: revsare = "last %d revs" % last_num
-    title("Success/Fail History of %s tests (%s)" % (type, revsare))
-
-
-#============================================================================================
-def plot_runtime(*args, **kwargs):
-    """ Call get_runtime_data()
-
-    Parameters
-    ----------
-        - See get_runtime_data() for the full list
-    """
-    (x,y) = get_runtime_data(*args, **kwargs)
-
-    figure()
-    index = np.arange(len(x))
-    plot(index,y,'-b.')
-    smart_ticks( index, x)
-    ylabel('Runtime/iteration (sec)')
-    xlabel(kwargs['x_field'])
-
-    last_num =kwargs.get('last_num',-1)
+# ============================================================================================
+def plot_runtime(annotate, saveImage, path, **kwargs):
+    name = kwargs['name']
+    (xData, yData) = get_runtime_data(**kwargs)
+    trace1 = go.Scatter(x=xData, y=yData,
+                        mode='lines+markers',
+                        marker=dict(
+                            size='5',
+                            color="blue"
+                        )
+                        )
+
+    annotations = []
+
+    last_num = kwargs.get('last_num', -1)
+
+    if annotate and not saveImage:
+        # retrieve commitids for annotation on the plotly graph
+        results = get_results(name, type='', where_clause='')
+        sorted = [(res["revision"], res["variables"], res["date"], res) for res in results]
+        sorted.sort(reverse=False)
```

Comment:
`sorted` is a built-in function for sorting (very similar to `[].sort` but a bit more general) and it would be better to avoid using it as a variable name.  Also, it looks like you're only using `revision`, `variables` and `date` for sorting by, and after that are only interested in `commitid`.

The sorting is probably better left to SQL, and `commitids` could then be generated using list comprehension, which could be implemented like this:

```python
    if annotate and not saveImage:
        # retrieve commitids for annotation on the plotly graph
        results = get_results(name, orderby_clause='ORDER BY revision, variables, date ASC')
        commitids = ["""<a href="%s/commit/%s">  </a>"""
                     % (MANTID_ADDRESS, res["commitid"])
                     for res in results]
```

It would be even better if `get_results` had an additional `field_names` parameter, so you could make it do `SELECT commitid FROM TestRuns WHERE ... ORDER BY ...` instead of `SELECT * FROM ...`, similar to how `get_all_field_values` does it. Just so that there's less data to retrieve and we get only exactly what we need. But that's probably not worthwhile here, the performance difference would be negligible I think.